source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm

```diff
@@ -0,0 +1,253 @@
+    carouselStackView.alignment = UIStackViewAlignmentCenter;
+    carouselStackView.translatesAutoresizingMaskIntoConstraints = NO;
+    carouselStackView.spacing = 20;
+    carouselStackView.clipsToBounds = YES;
+    
+    [carouselStackView addArrangedSubview:carouselPagesContainerView];
+    [carouselStackView addArrangedSubview:self.pageControl];
+    [carouselStackView addArrangedSubview:[[UIView alloc] initWithFrame:CGRectZero]];
+    
+    configBleed(rootView, elem, carouselViewContainer, acoConfig);
+    [self constructSwipeActions:carouselViewContainer];
+    
+    NSString *areaName = stringForCString(elem->GetAreaGridName());
+    [viewGroup addArrangedSubview:self withAreaName:areaName];
+    
+    [self addSubview:carouselViewContainer];
+    [NSLayoutConstraint activateConstraints:@[
+        [self.leadingAnchor constraintEqualToAnchor:carouselViewContainer.leadingAnchor],
+        [self.trailingAnchor constraintEqualToAnchor:carouselViewContainer.trailingAnchor],
+        [self.bottomAnchor constraintEqualToAnchor:carouselViewContainer.bottomAnchor],
+        [self.topAnchor constraintEqualToAnchor:carouselViewContainer.topAnchor]
+    ]];
+    
+    return self;
+}
+
+-(void) constructSwipeActions:(UIView *)view {
+    UISwipeGestureRecognizer *leftSwipe = [[UISwipeGestureRecognizer alloc]
+                                           initWithTarget:self
+                                           action:@selector(handleLeftSwipe:)];
+    leftSwipe.direction = UISwipeGestureRecognizerDirectionLeft;
+    [view addGestureRecognizer:leftSwipe];
+    
+    UISwipeGestureRecognizer *rightSwipe = [[UISwipeGestureRecognizer alloc]
+                                            initWithTarget:self
+                                            action:@selector(handleRightSwipe:)];
+    rightSwipe.direction = UISwipeGestureRecognizerDirectionRight;
+    [view addGestureRecognizer:rightSwipe];
+}
+
+- (void)handleLeftSwipe:(UISwipeGestureRecognizer *)gesture {
+    NSInteger newCarouselPageViewIndex = MIN(self.carouselPageViewIndex+1, self.carouselPageViewList.count-1);
+    if(newCarouselPageViewIndex == self.carouselPageViewIndex) {
+        return;
+    }
+    [self.pageControl setCurrentPage:newCarouselPageViewIndex];
+    
+    UIView *oldView = self.carouselPageViewList[self.carouselPageViewIndex];
+    UIView *newView = self.carouselPageViewList[newCarouselPageViewIndex];
+    
+    switch (self.carousel->getPageAnimation()) {
+        case PageAnimation::Slide:
+            [self slideAnimationForLeftSwipeForViewToHide:oldView
+                                   showView:newView];
+            break;
+        case PageAnimation::CrossFade:
+            [self crossfadeFromView:oldView toView:newView];
+        case PageAnimation::None:
+            oldView.hidden = YES;
+            newView.hidden = NO;
+        default:
+            break;
+    }
+    self.carouselPageViewIndex = newCarouselPageViewIndex;
```

Comment:
I think, This code(and similar below) should be part of carouselPagesContainerView?